src/main/java/io/xpire/logic/commands/TagCommand.java

```diff
@@ -111,12 +120,38 @@ public TagItemDescriptor(TagItemDescriptor toCopy) {
          * A defensive copy of {@code tags} is used internally.
          */
         public void setTags(Set<Tag> tags) {
-            this.tags = (tags != null) ? new HashSet<>(tags) : null;
+            if (tags != null) {
+                TreeSet<Tag> set = new TreeSet<>(new TagComparator());
+                set.addAll(tags);
+                this.tags = set;
+            } else {
+                this.tags = null;
+            }
         }
 
         public Set<Tag>getTags() {
             return (this.tags != null) ? Collections.unmodifiableSet(this.tags) : null;
         }
+
+        @Override
+        public boolean equals(Object other) {
+            // short circuit if same object
+            if (other == this) {
+                return true;
+            }
+
+            // instanceof handles nulls
+            if (!(other instanceof TagItemDescriptor)) {
+                return false;
+            }
+
+            // state check
+            TagItemDescriptor e = (TagItemDescriptor) other;
+
+            return getTags().equals(e.getTags());
+        }
```

Comment:
the .equals method format is different from DeleteCommand.java. Are we fixing on one format?

Reply:
I kinda based the TagCommand equals method on the EditCommand. 
For DeleteCommand I'm not very sure what should be the equal method. I think it should depend on whether we are deleting item/quantity/tags.